modules/engine/src/main/scala/observe/engine/Engine.scala

```diff
@@ -201,13 +213,15 @@ class Engine[F[_]: MonadThrow: Logger, S, U] private (
                 case Sequence.State.Final[F](_, _) => send(finished(id))
                 // Execution completed. Check breakpoint here
                 case _                             =>
-                  if (seq.getCurrentBreakpoint) {
+                  if (!isStarting && seq.getCurrentBreakpoint) {
                     switch(id)(SequenceState.Idle) *> send(breakpointReached(id))
                   } else
-                    switch(id)(SequenceState.Running(false, false, false)) *> send(executing(id))
+                    switch(id)(SequenceState.Running(false, false, false, false)) *> send(
```

Comment:
Shouldn't we do an `atomReload` when `isStarting` is true?

Reply:
No. `isStarting` is used to disable the breakpoint check in `atomReload`.